File: encryption.js
const crypto = require('crypto');
const key = process.env.ENCRYPTION_KEY;
if(!key){
  console.error('Missing ENCRYPTION_KEY');
  process.exit(1);
}
const IV_LENGTH = 16;
function encrypt(text){
  const iv = crypto.randomBytes(IV_LENGTH);
  const cipher = crypto.createCipheriv('aes-256-cbc', Buffer.from(key, 'hex'), iv);
  const encrypted = Buffer.concat([cipher.update(text, 'utf8'), cipher.final()]);
  return iv.toString('hex') + ':' + encrypted.toString('hex');
}
function decrypt(text){
  if(!text) return '';
  const parts = text.split(':');
  const iv = Buffer.from(parts[0], 'hex');
  const encryptedText = Buffer.from(parts[1], 'hex');
  const decipher = crypto.createDecipheriv('aes-256-cbc', Buffer.from(key, 'hex'), iv);
  const decrypted = Buffer.concat([decipher.update(encryptedText), decipher.final()]);
  return decrypted.toString('utf8');
}
module.exports = { encrypt, decrypt };

File: encryption.py
import os
from Crypto.Cipher import AES
from Crypto.Util import Padding

KEY = bytes.fromhex(os.getenv('ENCRYPTION_KEY', ''))
if not KEY:
    raise SystemExit('Missing ENCRYPTION_KEY')

def decrypt(value: str) -> str:
    if not value:
        return ''
    iv_hex, data_hex = value.split(':')
    iv = bytes.fromhex(iv_hex)
    data = bytes.fromhex(data_hex)
    cipher = AES.new(KEY, AES.MODE_CBC, iv)
    decrypted = cipher.decrypt(data)
    return Padding.unpad(decrypted, AES.block_size).decode('utf-8')
